qualang_tools/plot/Fitting/fitting_functions.py

```diff
@@ -0,0 +1,348 @@
+from scipy import optimize
+import numpy as np
+
+
+def fit_ramsey(x, y):
+    """
+    Create a fit to Ramsey experiment of the form
+
+    .. math::
+    f(x) = uncertainty_population * (1 - np.exp(-x * (1/tau))) + amp / 2 * (
+        np.exp(-x * (1/tau))
+        * (initial_offset * 2 + np.cos(2 * np.pi * f * x + phase))
+        )
+
+    for unknown parameters `f`, `phase`, `tau`, `amp`, `uncertainty_population` and `initial_offset`.
+
+    :param x:  The data on the x axis
+    :param y: The data on the y axis
```

Comment:
In what units should these be?

Reply:
In the linear fit it depends I think .. in Ramsey is it seconds? or ns ? as well as spectroscopy, is it Hz or MHz?

Reply:
For the linear case, I actually it is too general to put units.
For Ramsey I would put the dephasing time in ns and for the Y it depends on what data you expect to be there
For spectroscopy I would put Hz because we usually scan the frequency in Hz unit